src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -891,18 +894,20 @@ public Solution AddAnalyzerConfigDocuments(ImmutableArray<DocumentInfo> document
         /// </summary>
         public Solution RemoveDocument(DocumentId documentId)
         {
-            if (documentId == null)
-            {
-                throw new ArgumentNullException(nameof(documentId));
-            }
-
+            CheckContainsDocument(documentId);
             return RemoveDocuments(ImmutableArray.Create(documentId));
         }
 
         /// <summary>
         /// Creates a new solution instance that no longer includes the specified documents.
         /// </summary>
         public Solution RemoveDocuments(ImmutableArray<DocumentId> documentIds)
+        {
+            CheckContainsDocuments(documentIds);
+            return RemoveDocumentsImpl(documentIds);
+        }
+
+        private Solution RemoveDocumentsImpl(ImmutableArray<DocumentId> documentIds)
```

Comment:
Why the Impl method in this case? RemoveDocument isn't calling it...

Reply:
That's a bug. It should call it.